File: app/src/main/java/com/example/finext/MainActivity.java
package com.example.finext;

import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Gravity;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.example.finext.databinding.ActivityMainBinding;
import com.example.finext.fragments.BillpaymentFragment;
import com.example.finext.fragments.BudgetFragment;
import com.example.finext.fragments.ExpenseFragment;
import com.example.finext.fragments.MonthlyinsightsFragment;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        replaceFragment(new ExpenseFragment());

        binding.bottomNavigationView.setBackground(null);
        binding.bottomNavigationView.setOnItemSelectedListener(item -> {
            switch (item.getItemId()) {
                case R.id.expense:
                    replaceFragment(new ExpenseFragment());
                    break;
                case R.id.budgets:
                    replaceFragment(new BudgetFragment());
                    break;
                case R.id.bills:
                    replaceFragment(new BillpaymentFragment());
                    break;
                case R.id.insights:
                    replaceFragment(new MonthlyinsightsFragment());
                    break;
            }
            return true;
        });

        binding.fab.setOnClickListener(view -> showBottomDialog());
    }

    private void replaceFragment(Fragment fragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.frame_layout, fragment);
        fragmentTransaction.commit();
    }

    private void showBottomDialog() {
        final Dialog dialog = new Dialog(this);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setContentView(R.layout.bottomsheetlayout);

        LinearLayout videoLayout = dialog.findViewById(R.id.expense);
        LinearLayout shortsLayout = dialog.findViewById(R.id.budgets);
        LinearLayout liveLayout = dialog.findViewById(R.id.bills);
        //ImageView cancelButton = dialog.findViewById(R.id.cancelButton);

        videoLayout.setOnClickListener(v -> {
            dialog.dismiss();
            Toast.makeText(MainActivity.this, "Add an expense", Toast.LENGTH_SHORT).show();
        });

        shortsLayout.setOnClickListener(v -> {
            dialog.dismiss();
            Toast.makeText(MainActivity.this, "Add a budget", Toast.LENGTH_SHORT).show();
        });

        liveLayout.setOnClickListener(v -> {
            dialog.dismiss();
            Toast.makeText(MainActivity.this, "Add a bill", Toast.LENGTH_SHORT).show();
        });

        //cancelButton.setOnClickListener(view -> dialog.dismiss());

        dialog.show();
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.getWindow().getAttributes().windowAnimations = R.style.DialogAnimation;
        dialog.getWindow().setGravity(Gravity.BOTTOM);
    }
}

File: app/src/main/java/com/example/finext/MainActivity.kt
@file:Suppress("DEPRECATION")

package com.example.finext

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.navigation.compose.rememberNavController

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            ExpenseDashboard()
        }
    }

    @Preview
    @Composable
    fun ExpenseDashboard() {
        val navController = rememberNavController()

        Column(modifier = Modifier.padding(16.dp)) {
            Text(text = "Total Expense: $500")
            Row {
                Button(
                    onClick = {
                        navController.navigate("expenseFragment")
                    },
                    modifier = Modifier.padding(top = 16.dp)
                ) {
                    Text(text = "Add Expense")
                }

                Button(
                    onClick = {
                        navController.navigate("budgetFragment")
                    },
                    modifier = Modifier.padding(top = 16.dp)
                ) {
                    Text(text = "Create Budget")
                }
            }
            Row {
                Button(
                    onClick = {
                        navController.navigate("billPaymentFragment")
                    },
                    modifier = Modifier.padding(top = 16.dp)
                ) {
                    Text(text = "Bill Payment")
                }
                Button(
                    onClick = {
                        // Code to navigate to Monthly Insights screen
                    },
                    modifier = Modifier.padding(top = 16.dp)
                ) {
                    Text(text = "Monthly Insights")
                }
            }
        }
    }


    @Deprecated("Deprecated in Java")
    override fun onBackPressed() {
        if (supportFragmentManager.backStackEntryCount > 0) {
            supportFragmentManager.popBackStack()
        } else {
            super.onBackPressed()
        }
    }
}
